test/upload-test-images.sh

```diff
@@ -19,28 +19,37 @@ set -o errexit
 # If gcloud is not available make it a no-op, not an error.
 which gcloud &> /dev/null || gcloud() { echo "[ignore-gcloud $*]" 1>&2; }
 
+function upload_single_test_image() {
+  local docker_tag=$1
+  local image_name=$2
+  local image="knative.dev/eventing/test/test_images/${image_name}"
+  local registry=${KO_DOCKER_REPO%/*}
+
+  ko publish -B ${image}
+
+  if [ -n "$docker_tag" ]; then
+      if [ "$registry" = "docker.io" ]; then
+          ko publish -B -t ${docker_tag} ${image}
```

Comment:
You could do like serving does, and use `ko resolve -R` to upload all images at once without needing a loop or background processes. The following benefits would also be included: upload errors will be easily spotted; works also with gcr.io.

Reply:
Yeah I was looking into the one in serving, and it seems we can simply copy and paste here..

Reply:
Hmm, adding a yaml for every image seems to be over complicated for us, since we don't actually need them in eventing test.
@adrcunha I feel this solution is not as good as the first one, except I'm not quite certain about your comment "upload errors will be easily spotted", can you elaborate it more?

Reply:
I disagree. The yamls are simple and are just definitions of the images. They could even be autogenerated, and I think they should have the boilerplate and a one-time generic comment about their existence, but I won't block this PR on those items.

The initial solution is more complicated in terms of code, this one is way cleaner and leverage existing tools. Regarding the errors, I mean that spawning several commands in background turns out to be a pain if something fails.

/lgtm